plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -60,73 +63,82 @@
 import java.io.IOException;
 import java.util.*;
 
-public class PipelineStore extends AbstractLifecycleComponent {
+public class PipelineStore extends AbstractComponent implements Closeable {
 
     public final static String INDEX = ".ingest";
     public final static String TYPE = "pipeline";
 
-    private final ThreadPool threadPool;
     private final TimeValue scrollTimeout;
-    private final ClusterService clusterService;
     private final Provider<Client> clientProvider;
-    private final TimeValue pipelineUpdateInterval;
     private final Pipeline.Factory factory = new Pipeline.Factory();
     private final Map<String, Processor.Factory> processorFactoryRegistry;
 
     private volatile Client client;
+    private volatile boolean pipelineIndexReady = false;
     private volatile Map<String, PipelineDefinition> pipelines = new HashMap<>();
 
     @Inject
-    public PipelineStore(Settings settings, Provider<Client> clientProvider, ThreadPool threadPool, Environment environment, ClusterService clusterService, Map<String, ProcessorFactoryProvider> processorFactoryProviders) {
+    public PipelineStore(Settings settings, Provider<Client> clientProvider, Environment environment, Map<String, ProcessorFactoryProvider> processorFactoryProviders) {
         super(settings);
-        this.threadPool = threadPool;
-        this.clusterService = clusterService;
         this.clientProvider = clientProvider;
         this.scrollTimeout = settings.getAsTime("ingest.pipeline.store.scroll.timeout", TimeValue.timeValueSeconds(30));
-        this.pipelineUpdateInterval = settings.getAsTime("ingest.pipeline.store.update.interval", TimeValue.timeValueSeconds(1));
         Map<String, Processor.Factory> processorFactories = new HashMap<>();
         for (Map.Entry<String, ProcessorFactoryProvider> entry : processorFactoryProviders.entrySet()) {
             Processor.Factory processorFactory = entry.getValue().get(environment);
             processorFactories.put(entry.getKey(), processorFactory);
         }
         this.processorFactoryRegistry = Collections.unmodifiableMap(processorFactories);
-        clusterService.add(new PipelineStoreListener());
     }
 
     @Override
-    protected void doStart() {
-    }
-
-    @Override
-    protected void doStop() {
-    }
-
-    @Override
-    protected void doClose() {
-        // TODO: When org.elasticsearch.node.Node can close Closable instances we should remove this code
+    public void close() throws IOException {
+        // TODO: When org.elasticsearch.node.Node can close Closable instances we should try to remove this code,
```

Comment:
is there a reference issue to keep track of this to-be-introduced functionality?